pkg/azure/blob.go

```diff
@@ -51,6 +51,11 @@ const (
 	clientCertificateField         = "clientCertificate"
 	clientCertificatePasswordField = "clientCertificatePassword"
 	accountKeyField                = "accountKey"
+
+	// Ref: https://docs.microsoft.com/en-us/azure/aks/kubernetes-service-principal?tabs=azure-cli#manually-create-a-service-principal
+	tenantField   = "tenant"
```

Comment:
I don't think this is needed.
To my knowledge, the AKS service principal is not stored as `Secret` in Kubernetes so the user may still need to create a new `Secret` and they can always use the `clientId`, `cilentSecret` and `tenantId`.

Reply:
The reason I added this is because we have another Azure implementation pending in https://github.com/fluxcd/kustomize-controller/pull/495, which does have a valid use-case for this field (as the data is loaded from a Secret key containing a _file_). By aligning them, we hopefully help reduce users their cognitive load.